cp_ws_helpers.py

```diff
@@ -0,0 +1,21 @@
+import re
+
+# Helper functions for dealing conforming to the Bioleap WS Apps message protocol
+# which send packets like `<requestid> <cmdname> <message data>` in both directions.
+
+def extractWsAppMessage(wholeMessage):
+    """
+    Bioleap WsApps messages look like `<requestid> <cmdname> <message data>`
+    """
+    match = re.match("(#[a-z-_]+\d+) ([a-z-_]+) (\{.*\})", wholeMessage, re.DOTALL)
```

Comment:
I think this wants to be an "r" (raw) string. Otherwise the backslashes are taken as python string escapes and won't get into the re. Simply insert an "r" before the string.

Reply:
Yeah, this is good. That function is general and doesn't need to expect json data in the message content.